network/p2p/scoring/registry.go

```diff
@@ -133,6 +136,25 @@ func NewGossipSubAppSpecificScoreRegistry(config *GossipSubAppSpecificScoreRegis
 		idProvider:     config.IdProvider,
 	}
 
+	builder := component.NewComponentManagerBuilder()
+	builder.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+		reg.logger.Info().Msg("starting subscription validator")
+		reg.validator.Start(ctx)
+		select {
+		case <-ctx.Done():
+			reg.logger.Warn().Msg("aborting subscription validator startup, context cancelled")
+		case <-reg.validator.Ready():
+			reg.logger.Info().Msg("subscription validator started")
+		}
+		ready()
```

Comment:
I assume the worker isn't `ready()` unless the validator is
```suggestion
		case <-reg.validator.Ready():
			reg.logger.Info().Msg("subscription validator started")
			ready()
		}
```

Reply:
https://github.com/onflow/flow-go/pull/4988/commits/48f255fffa25c9f0ae94d250a6ef21d3c874e9f4